### ISSUE TITLE
Improve page load performance

### ISSUE THREAD

**Issue by josueetcom** [beyondhb1079/s4us#1102]
[PageSpeed Insights](https://pagespeed.web.dev/report?url=https%3A%2F%2Fdreamscholars.org%2F) rates our site at 90/100 for Desktop performance and 36/100 for mobile performance:

<img width="994" alt="Screen Shot 2022-05-11 at 8 06 38 PM" src="https://user-images.githubusercontent.com/8023233/167983979-19ee39d8-be7e-4a21-8254-3f0f164b855b.png">

We should really address performance issues by leveraging code-splitting (lazy loading) and Suspense:
https://web.dev/code-splitting-suspense/

**Comment by josueetcom** [comment]
For reference, here is the score after #1108 but before #1114:
<img width="1022" alt="Screen Shot 2022-05-14 at 3 13 41 AM" src="https://user-images.githubusercontent.com/8023233/168421478-04cb8217-310a-4f89-be1d-7b1720f1bd63.png">

Big gains in FCP and SI scores! #1114 should hopefully address LCP. Still gotta figure out how to improve TI and TBT :P

**Comment by josueetcom** [comment]
Looks like we have a better score now after merging those two PRs:

<img width="1012" alt="Screen Shot 2022-05-15 at 1 03 26 AM" src="https://user-images.githubusercontent.com/8023233/168463243-8217b232-cbc1-40b3-8e04-6b9a046c3046.png">

TTI, LCP and TBT have dropped without affecting FCP and SI

**Comment by josueetcom** [comment]
What remains to be improved has to do with the JS we send to the browser. There's a lot of unused code:
<img width="1434" alt="Screen Shot 2022-05-15 at 1 14 35 AM" src="https://user-images.githubusercontent.com/8023233/168463590-e1883a91-294e-4f75-a702-ef08f891d9cc.png">

Unfortunately most of it is has to do with Firebase:
- 15% (175KiB) is for the Firestore library. Loaded for ScholarshipsContext. Not needed for the public home page but is needed for other pages.
- 14% is for the analytics library (green box at the bottom)
- And the `iframe.js` file somewhere is for Firebase Auth. It doesn't always appear but on PageSpeed it says it's 263 KiB and not zipped. In practice though the mobile view in my inspector gives around ~86 KiB and it _is_ gzipped.
- Everything else there is pretty much needed as far as I can tell.

But, these _are_ some things we could try:
- Some how code splitting away ScholarshipsContext and its firestore imports so that they're only used on an as-needed basis (e.g. under the `/scholarships` routes). Nested routes might make sense. This would address 15% of the bundle size and some of the blocking time for pages that don't use the context.
- Use [Firestore Lite](https://firebase.google.com/docs/firestore/solutions/firestore-lite) instead? We'd be missing some features like caching but we could always change it back later (much easier to go from lite -> full than the other way).
- [Deploy ES2015](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/) to browsers that support it. Apparently the reduction is about 50% or more! Which would be totally worth it, especially as we add more features to the app.

The last one seems the most promising to me, especially since every major browser supports it since March 2017:
<img width="977" alt="Screen Shot 2022-05-15 at 1 38 51 AM" src="https://user-images.githubusercontent.com/8023233/168464369-755cb129-0975-4da5-a2f9-fdfb705d669a.png">
 
It looks like at least [95.46%](https://caniuse.com/?search=es6) of browser versions used in the US fully support ES6. Of the ones tracked, 98.36%. And of mobile users specifically, 99.15%.  Definitely seems like an easy win to me!

**Comment by josueetcom** [comment]
Next step: follow [this guide](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/) to send modern JavaScript.

**Comment by josueetcom** [comment]
According to https://estimator.dev/#dreamscholars.org modern JS might only make the site around 12% faster :P
